Add resolver for Twitch users

M4tthewDE opened the pull request:
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Implements a resolver for Twitch user links.

Fixes #132

pajlada commented on pkg/config/model.go:
```diff
@@ -11,7 +11,6 @@ type APIConfig struct {
 	EnableAnimatedThumbnails bool   `mapstructure:"enable-animated-thumbnails" json:"enable-animated-thumbnails"`
 	MaxThumbnailSize         uint   `mapstructure:"max-thumbnail-size" json:"max-thumbnail-size"`
 
-	TwitchUsernameCacheDuration      time.Duration `mapstructure:"twitch-username-cache-duration" json:"twitch-username-cache-duration"`
```
We will want to add a changelog entry noting that this is no longer used, or some runtime warning, before just removing it

M4tthewDE replied:
I feel like it makes more sense to just use this config entry since it already exists. I added an entry to the example config.yaml to document it.

pajlada reviewed (changes requested):
1. Usernames should be lowercased before attempting to be loaded
If I type `https://twitch.tv/pajlada` and `https://twitch.tv/PAJLADA` they will be cached separately
2. The information given feels really bare, I'd like it if there was some sort of indication that this was information about the Twitch user
3. The template works well for users with a different login name & display name, but looks ugly when the user has the same login name & display name. We should do a case-insensitive comparison of the login name & the display name. If they are equal, use only the display name. If they are not equal, use the current format: `display name (login name)`
**Examples:**  
Login name=`pajlada`, Display Name=`pajlada`. Should show `pajlada`  
Login name=`testaccount_420`. Display Name=`테스트계정420`. Should show `테스트계정420 (testaccount_420)`  
Login name=`gazatu`. Display Name=`GaZaTu`. Should show `GaZaTu`


Separately, imo not for this PR: We could lower the default cache time & include their live status if they're streaming. I think that would be neat

M4tthewDE commented:
1. Fixed.
2. I added an indication that the user is on Twitch and added the URL leading to their profile. This is more in line with what the OpenGraph tags return. Note that this PR adds a slight regression, since the OG tags usually show the stream title in case the channel this live. This can be improved in a follow up PR as you said.
3. Fixed

pajlada reviewed (changes requested):
There are some usernames we "need" to manually ignore (i.e. not resolve or handle at all):
 - `inventory`
 - `popout`
 - `subscriptions`
 - `videos`
